### PR TITLE
Add hanmanwuzhe source

### LINKED ISSUE #5086: Add source: 韩漫无遮挡 / hanmanwuzhe
### Source name

韩漫无遮挡（hanmanwuzhe）

### Source link

https://www.hanmanwuzhe.com

### Source language

Chinese

### Other details

18+/NSFW = yes

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by heagan01**
Closes #5086 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+    override fun imageRequest(page: Page): Request {
+        val newHeaders = headersBuilder()
+            .set("Accept", "image/avif,image/webp,image/png,image/jpeg,*/*")
+            .build()
+
+        return GET(page.imageUrl!!, newHeaders)
+    }
+
+    private class Genre(name: String, val id: String = name) : Filter.CheckBox(name)
```
Comment (anenasa) [inline src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: I have a question, why use CheckBox if only one genre can be chosen?

Comment (heagan01) [reply src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: I don't really know the functions, sorry 😅, is there a function for a radio button?

Comment (anenasa) [reply src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: You can use Filter.Select to create drop-down list.

Comment (heagan01) [reply src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: ok, will try

Comment (heagan01) [reply src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: is this alright?

Comment (anenasa) [reply src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: Emmm, I would suggest using Pair so you can put genre name and url in the same array. You can read code of other extensions, like [manhuaren](https://github.com/keiyoushi/extensions-source/blob/83a2c48ef5f4ec5162c2b60541c9adacd0421f8b/src/zh/manhuaren/src/eu/kanade/tachiyomi/extension/zh/manhuaren/Manhuaren.kt#L590C6-L590C47) or [zerobyw](https://github.com/keiyoushi/extensions-source/blob/83a2c48ef5f4ec5162c2b60541c9adacd0421f8b/src/zh/zerobyw/src/eu/kanade/tachiyomi/extension/zh/zerobyw/Zerobyw.kt#L153)

**Comment by stevenyomi** [comment]
This site is MCCMS, I'll review later.

**Comment by heagan01** [comment]
are there any updates for this? @stevenyomi

**Comment by AwkwardPeak7** [comment]
@heagan01 you should move this to MCCMS multisrc as steven pointed out

**Comment by heagan01** [comment]
I've done it, are there any changes needed?

File: src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class hanmanwuzhe : MCCMS(
+    "嘀嗒漫画",
+    "https://www.hanmanwuzhe.com",
+    "zh",
+    MCCMSConfig(useMobilePageList = true),
+) {
+    // Details and chapter pages are broken
+    override fun getMangaUrl(manga: SManga) = baseUrl
+    override fun getChapterUrl(chapter: SChapter) = baseUrl
```
Comment (AwkwardPeak7) [inline src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: broken how? I just tested and should work fine

Comment (heagan01) [reply src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]: i copy pasted the code from other extension (DidaManhua)

Comment (AwkwardPeak7) [inline src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]:
```suggestion
```

File: src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package eu.kanade.tachiyomi.extension.zh.hanmanwuzhe
+
+import eu.kanade.tachiyomi.multisrc.mccms.MCCMS
+import eu.kanade.tachiyomi.multisrc.mccms.MCCMSConfig
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
```
Comment (AwkwardPeak7) [inline src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt]:
```suggestion
```

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**